File: 412.fizz-buzz.java
import java.util.ArrayList;

/*
 * @lc app=leetcode.cn id=412 lang=java
 *
 * [412] Fizz Buzz
 */

// @lc code=start
class Solution {
    public List<String> fizzBuzz(int n) {
        List<String> res = new ArrayList<>();

        for (int i = 1; i <= n; ++i) {
            StringBuilder ans = new StringBuilder();

            if (i % 3 == 0) {
                ans.append("Fizz");
            }
            if (i % 5 == 0) {
                ans.append("Buzz");
            }

            if (ans.length() == 0) {
                res.add(Integer.toString(i));
            } else {
                res.add(ans.toString());
            }
        }
        return res;
    }
}
// @lc code=end

File: 412.fizz-buzz.cpp
/*
 * @lc app=leetcode id=412 lang=cpp
 *
 * [412] Fizz Buzz
 */
#include <bits/stdc++.h>

// @lc code=start
class Solution {
 public:
  std::vector<std::string> fizzBuzz(int n) {
    std::vector<std::string> res;
    for (int i = 1; i <= n; ++i) {
      std::string ans;

      if (i % 3 == 0) {
        ans += "Fizz";
      }
      if (i % 5 == 0) {
        ans += "Buzz";
      }
      if (ans.empty()) {
        res.push_back(std::to_string(i));
      } else {
        res.push_back(std::move(ans));
      }
    }
    return res;
  }
};
// @lc code=end
